cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -766,19 +766,19 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
                             if (featureGates.useStrimziPodSetsEnabled())   {
                                 return strimziPodSetOperator.getAsync(namespace, KafkaResources.kafkaStatefulSetName(name))
                                         .compose(podSet -> {
-                                            if (podSet != null)    {
-                                                return Future.succeededFuture(podSet.getSpec().getPods().size());
+                                            if (podSet != null) {
+                                                return Future.succeededFuture(KafkaCluster.generatePodList(reconciliation.name(), podSet.getSpec().getPods().size()));
                                             } else {
-                                                return Future.succeededFuture(0);
+                                                return Future.succeededFuture(new ArrayList<String>());
```

Comment:
`List.of()`

Reply:
List.of() will return me a list of object type(List) while the generatePodList podList method is returning me a List of string type(List<String) and since this if-else is later on followed by a compose so the future returned should be of one type either List<E> or List<String>

Reply:
`List.<String>of()` then if type arg inference doesn't work.